### ISSUE TITLE
Can't build a package with features when feature name contains hypens

### ISSUE THREAD

**Issue by Zaharid** [conda/conda-build#1019]
Consider a package that contains only  the following meta.yaml:

```
package:
  # This name doesn't have to be the same as the feature, but can avoid confusion if it is
  name: x_aaa
  # This need not relate to the version of any of the packages with the
  # feature. It is just a version for this metapackage.
  version: 1.0

build:
  features:
      - x-aaa
```

It fails with the following traceback:

```
Removing old build environment
BUILD START: x_aaa-1.0-x-aaa_0
Package: x_aaa-1.0-x-aaa_0
no source
number of files: 0
Fixing permissions
Fixing permissions
BUILD END: x_aaa-1.0-x-aaa_0
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda-build/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/home/zah/anaconda3/bin/conda-build", line 5, in <module>
    sys.exit(main())
  File "/home/zah/anaconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 134, in main
    args_func(args, p)
  File "/home/zah/anaconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 373, in args_func
    args.func(args, p)
  File "/home/zah/anaconda3/lib/python3.5/site-packages/conda_build/main_build.py", line 320, in execute
    need_source_download=need_source_download)
  File "/home/zah/anaconda3/lib/python3.5/site-packages/conda_build/build.py", line 571, in build
    tarcheck.check_all(path)
  File "/home/zah/anaconda3/lib/python3.5/site-packages/conda_build/tarcheck.py", line 60, in check_all
    x.index_json()
  File "/home/zah/anaconda3/lib/python3.5/site-packages/conda_build/tarcheck.py", line 53, in index_json
    getattr(self, varname)))
Exception: name: 'x_aaa' != 'x_aaa-1.0'
```

It works if I change the feature name to not contain hyphens. The version of conda build is:

```
$ conda build --version -
conda-build 1.20.3
```

**Comment by pelson** [comment]
It would be helpful if conda-build asserted that there were no `-` characters in the feature name. I'd probably look at putting this around https://github.com/conda/conda-build/blob/master/conda_build/metadata.py#L179. Interested in having a go @Zaharid?

**Comment by dfroger** [comment]
The issue has been fixed in #1600

**Comment by msarahan** [comment]
Thanks for the update.  Closing.
